includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -790,32 +790,24 @@ private function is_product_supported() {
 		$product      = $this->get_product();
 		$is_supported = true;
 
-		if ( ! is_object( $product ) || ! in_array( $product->get_type(), $this->supported_product_types(), true ) ) {
+		if ( is_null( $product ) ) {
 			$is_supported = false;
-		}
-
-		// Trial subscriptions with shipping are not supported.
-		if ( class_exists( 'WC_Subscriptions_Product' ) && $product->needs_shipping() && WC_Subscriptions_Product::get_trial_length( $product ) > 0 ) {
+		} elseif ( ! is_object( $product ) || ! in_array( $product->get_type(), $this->supported_product_types(), true ) ) {
```

Comment:
What do you think about refactoring the whole if-else ladder as:

```php
if ( is_null( $product )
  || ! is_object( $product )
  || ! $this->is_supported_type( $product )
  || $this->is_trial_subscription_with_shipping( $product )
  || $this->product_is_charged_upon_release( $product )
  ...
 ) {
  $is_supported = false;
}
```

Reply:
Did we add the `is_null` check only because we changed the default return to `null` in `get_product`?

> But I'd check once with @ricardo as for why get_product is currently returning false by default (https://github.com/Automattic/woocommerce-payments/pull/1584).

I think it's because `wc_get_product` will return `false` by default, so I just followed the same pattern.